presto-main/src/main/java/io/prestosql/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -231,6 +237,11 @@ public void enqueue(List<SerializedPage> pages)
 
         // drop the initial reference
         serializedPageReferences.forEach(SerializedPageReference::dereferencePage);
+
+        // if the buffer is full for first time, update the task status
+        if (!hasBlockedBefore.get() && !isFull().isDone() && hasBlockedBefore.compareAndSet(false, true)) {
```

Comment:
should we check for `isFull` or the logic we use at coordinator (> 50% util.)?

Reply:
Let's use `isFull`, 50% condition is opportunistic when there is no notification